pkg/index.d.ts

```diff
@@ -58,6 +58,8 @@ export type Message =
       }
     >
   | BaseMessage<'EXPORTS_TYPES_SHOULD_BE_FIRST'>
+  | BaseMessage<'EXPORTS_MODULE_SHOULD_PRECEED_IMPORT'>
+  | BaseMessage<'EXPORTS_MODULE_SHOULD_PRECEED_REQUIRE'>
```

Comment:
Thanks for the PR! This feature makes sense to me.

I think we can merge this two as one, e.g. `EXPORTS_MODULE_SHOULD_PRECEED_IMPORT_REQUIRE`, since they're quite similar. And add a `conditions` param to represent `import` and `require`.

So something like
```suggestion
  | BaseMessage<'EXPORTS_MODULE_SHOULD_PRECEED_IMPORT_REQUIRE', { conditions: string[] }>
```